src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1524,6 +1524,25 @@ bool ParserNull::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
         return false;
 }
 
+static bool parseNumber(char * buffer, size_t size, bool negative, int base, Field & res)
+{
+    errno = 0;    /// Functions strto* don't clear errno.
+
+    char * pos_integer = buffer;
```

Comment:
Do we need to set `char * pos_integer = buffer` ? Seems like it's possible to make `buffer` const
```suggestion
static bool parseNumber(const char * buffer, size_t size, bool negative, int base, Field & res)
{
    errno = 0;    /// Functions strto* don't clear errno.

    char * pos_integer;
```

Reply:
It is required to be char * because `pos_integer` is required to be mutable to pass into `strtoull`.

Reply:
`str_end` is output-only parameter of `strtoull`, isn't it? So we don't have to initialize `pos_integer`